public/app/plugins/panel/heatmap-new/ColorScale.tsx

```diff
@@ -10,55 +10,69 @@ type Props = {
+  hoverValue?: number;
 };
 
 type HoverState = {
   isShown: boolean;
   value: number;
 };
 
-export const ColorScale = ({ colorPalette, min, max, display }: Props) => {
+export const ColorScale = ({ colorPalette, min, max, display, hoverValue }: Props) => {
   const [colors, setColors] = useState<string[]>([]);
-  const [hover, setHover] = useState<HoverState>({ isShown: false, value: 0 });
-  const [cursor, setCursor] = useState({ clientX: 0, clientY: 0 });
+  const [scaleHover, setScaleHover] = useState<HoverState>({ isShown: false, value: 0 });
+  const [percent, setPercent] = useState<number | null>(null);
+
+  const theme = useTheme2();
+  const styles = getStyles(theme, colors);
 
   useEffect(() => {
     setColors(getGradientStops({ colorArray: colorPalette }));
   }, [colorPalette]);
 
-  const theme = useTheme2();
-  const styles = getStyles(theme, colors);
-
   const onScaleMouseMove = (event: React.MouseEvent<HTMLDivElement>) => {
     const divOffset = event.nativeEvent.offsetX;
     const offsetWidth = (event.target as any).offsetWidth as number;
     const normPercentage = Math.floor((divOffset * 100) / offsetWidth + 1);
     const scaleValue = Math.floor(((max - min) * normPercentage) / 100 + min);
-    setHover({ isShown: true, value: scaleValue });
-    setCursor({ clientX: event.clientX, clientY: event.clientY });
+
+    setScaleHover({ isShown: true, value: scaleValue });
+    setPercent(normPercentage);
   };
 
   const onScaleMouseLeave = () => {
-    setHover({ isShown: false, value: 0 });
+    setScaleHover({ isShown: false, value: 0 });
   };
 
+  useEffect(() => {
+    if (hoverValue != null) {
+      const percent = hoverValue / (max - min);
+      setPercent(percent * 100);
+    }
+  }, [hoverValue, min, max]);
+
   return (
     <div className={styles.scaleWrapper}>
-      <div>
-        <div className={styles.scaleGradient} onMouseMove={onScaleMouseMove} onMouseLeave={onScaleMouseLeave}>
-          {display && hover.isShown && (
-            <VizTooltipContainer position={{ x: cursor.clientX, y: cursor.clientY }} offset={{ x: 10, y: 10 }}>
-              {display(hover.value)}
-            </VizTooltipContainer>
-          )}
-        </div>
-        {display && (
-          <div>
-            <span>{display(min)}</span>
-            <span className={styles.maxDisplay}>{display(max)}</span>
+      <div className={styles.scaleGradient} onMouseMove={onScaleMouseMove} onMouseLeave={onScaleMouseLeave}>
+        {display && (scaleHover.isShown || hoverValue !== undefined) && (
+          <div className={styles.followerContainer}>
+            <div className={styles.follower} style={{ left: percent + '%' }} />
           </div>
         )}
       </div>
+      {display && (
+        <div className={styles.followerContainer}>
+          {percent != null && (scaleHover.isShown || hoverValue !== undefined) && (
+            <span className={styles.hoverValue} style={{ left: percent - 2 + '%' }}>
```

Comment:
Might be worth for as a maintainability standpoint to extract these numbers as constants e.g.
```suggestion
// Top of file
const LEFT_OFFSET = 2

            <span className={styles.hoverValue} style={{ left: `${percent - LEFT_OFFSET} %`}}>
```

Similar for the opacity number down below as well (i.e. `0.3` + `1`)